src/Interpreters/HashJoin/HashJoin.cpp

```diff
@@ -653,9 +663,10 @@ bool HashJoin::addBlockToJoin(ScatteredBlock & source_block, bool check_limits)
         }
 
         doDebugAsserts();
-        data->blocks_allocated_size += block_to_save.allocatedBytes();
-        data->blocks.emplace_back(std::move(block_to_save));
-        const auto * stored_block = &data->blocks.back();
+        size_t data_allocated_bytes = block_to_save.allocatedBytes();
```

Comment:
here we seem to use the specialised `ScatteredBlock::allocatedBytes()` too

Reply:
yes, I think it was intended to calculate only scattered data. Also, it looks like block_to_save did not have selector, so maybe it would change the behavior. Let's try :)